apps/snitch_core/lib/core/domain/taxonomy/taxonomy.ex

```diff
@@ -275,7 +287,47 @@ defmodule Snitch.Domain.Taxonomy do
   @doc """
   Delete a taxon
   """
-  def delete_taxon(taxon) do
-    taxon |> AsNestedSet.delete() |> AsNestedSet.execute(Repo)
+  def delete_taxon(taxon_id) do
+    case get_taxon(taxon_id) do
+      %Taxon{} = taxon ->
+        Multi.new()
+        |> Multi.run(:delete_products, fn _ -> do_delete_products(taxon_id) end)
+        |> Multi.run(:category, fn _ -> do_delete_taxon(taxon) end)
+        |> Repo.transaction()
+
+      nil ->
+        {:error, :not_found}
+    end
+  end
+
+  defp do_delete_products(taxon_id) do
+    categories =
```

Comment:
can be extracted to product model. with success and error return. which will let the multi move ahead.